changelog.d/12939.feature

```diff
@@ -0,0 +1 @@
+MSC3786: Check `state_key`.
```

Comment:
I'd throw in some extra words for clarity:
```suggestion
Update MSC3786 implementation to check `state_key`.
```

Reply:
Also -- I think this is a bugfix, not a feature?